ocaml/typing/typemod.ml

```diff
@@ -503,21 +503,24 @@ let check_well_formed_module env loc context mty =
 let () = Env.check_well_formed_module := check_well_formed_module
 
 let type_decl_is_alias sdecl = (* assuming no explicit constraint *)
+  let eq_vars x y =
+    match Jane_syntax.Core_type.of_ast x, Jane_syntax.Core_type.of_ast y with
+    | Some (Jtyp_layout (Ltyp_var { name = Some nx; layout = lx }), _attrsx),
+      Some (Jtyp_layout (Ltyp_var { name = Some ny; layout = ly }), _attrsy) ->
+      String.equal nx ny &&
+      Location.compare_txt Layout.equal_const lx ly
+    | (Some _, _) | (_, Some _) -> false
```

Comment:
What if the layout annotation constrains the layout, though? Something like

```
type ('a : value) t1
type ('a : immediate) t2 = 'a t1
```

That's not an alias, yet it meets your standard. Yet

```
type ('a : immediate) t3 = 'a t2
```

*is* an alias. The only way we can know is by looking in the env. That's hard to do here. So let's not.

Please resolve if you agree with this analysis, which agrees that the current implementation is over-conservative, but that it's hard to do better. I'll also add a comment to the code.

Reply:
Actually, I think I've just convinced myself that the status quo is wrong. Because my implementation will also return true for

```
type ('a : immediate) t4 = ('a : immediate) t
```

which is *also* not an alias.

Reply:
Will push new code here shortly. Please check it over and Resolve if happy. Thanks.

Reply:
(For future reference: I think I was making a comment about how this function is used, and Richard was thinking about what its behavior should be based on its name, but either way is fine in this case and if this does turn out to be conservative for some particular use case we can adjust it later.)